### PR TITLE
MARP-255 Missing Portal version 8 Cannot download from market

### PR DESCRIPTION
**Pull request by phhung-axonivy**
Missing Portal version 8.x - Cannot see to download from market

### CODE REVIEW

File: src/app/domain/maven/MavenArtifact.php [hunk 1]
```diff
@@ -164,23 +164,31 @@ private function getBaseUrlFromGroupIdAndArtifactId($targetGroupId, $artifactId)
...
+    if (!empty($xml)) {
+      $v = self::parseVersions($xml);
       usort($v, 'version_compare');
       $v = array_reverse($v);
       $v = self::filterSnapshotsWhichAreRealesed($v);
-      $this->versionCache = $v;
+      if ($this->versionCache == null) {
+        $this->versionCache = $v;
+      } else {
+        $this->versionCache = array_merge($this->versionCache, $v);
```
Comment (alexsuter) [inline src/app/domain/maven/MavenArtifact.php]: you should sort the versionCache after merging the array

File: src/app/domain/maven/MavenArtifact.php [hunk 2]
```diff
@@ -164,23 +164,31 @@ private function getBaseUrlFromGroupIdAndArtifactId($targetGroupId, $artifactId)
...
     if ($this->versionCache == null) {
-      $baseUrl = $this->getBaseUrl();
-
-      $xml = HttpRequester::request("$baseUrl/maven-metadata.xml");
-
-      if (empty($xml)) {
-        $this->versionCache = [];
-        return $this->versionCache;
+      foreach ($this->archivedArtifacts as $archivedArtifact) {
+        $archivedArtifactBaseUrl = $this->getBaseUrlFromGroupIdAndArtifactId($archivedArtifact->getGroupId(), $archivedArtifact->getArtifactId());
```
Comment (alexsuter) [inline src/app/domain/maven/MavenArtifact.php]: I would prefer to see a method which calculates the value for $this->versionCache

**Comment by phhung-axonivy** [comment]
hi @alexsuter 
Thanks for your feedback.
I handled your feedback. Could you help me review the code again?

File: src/app/domain/maven/MavenArtifact.php [hunk 3]
```diff
@@ -164,23 +164,52 @@ private function getBaseUrlFromGroupIdAndArtifactId($targetGroupId, $artifactId)
...
+      return;
+    }
+
+    if ($this->versionCache != null) {
+      $versions = array_merge($this->versionCache, $versions);
+      usort($versions, 'version_compare');
+      $versions = array_reverse($versions);
+    }
+
+    $this->versionCache = $versions;
```
Comment (alexsuter) [inline src/app/domain/maven/MavenArtifact.php:212]: versionCache will be not sorted if this method is only called once with $versions != null

Comment (phhung-axonivy) [reply src/app/domain/maven/MavenArtifact.php:212]: $version is always sorted in the current implementation but I will change the code to improve maintainability

**Review by alexsuter — APPROVED** [review]

**Pull request merged**